src/lib/optimizer/optimizer.cpp

```diff
@@ -101,11 +106,6 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
 
   optimizer->add_rule(std::make_unique<BetweenCompositionRule>());
 
-  // Run the ColumnPruningRule before the PredicatePlacementRule, as it might turn joins into semi joins, which
-  // can be treated as predicates and pushed further down. For the same reason, run it after the JoinOrderingRule,
-  // which does not like semi joins (see above).
-  optimizer->add_rule(std::make_unique<ColumnPruningRule>());
-
   // Position the predicates after the JoinOrderingRule ran. The JOR manipulates predicate placement as well, but
   // for now we want the PredicateReorderingRule to have the final say on predicate positions
```

Comment:
Trailing dot.
And I don't get the connection between running the `PredicatePlacementRule` a first time and "the PredicateReorderingRule to have the final say on predicate positions".

The argument for the second run is clear to me.